src/cmd/services/m3coordinator/downsample/id_pool_types.go

```diff
@@ -215,6 +218,25 @@ func (p *rollupIDProvider) Duplicate() ident.TagIterator {
 	return duplicate
 }
 
+func (p *rollupIDProvider) Hash() (uint32, error) {
+	if len(p.tagPairs) == 0 {
+		return 0, nil
+	}
+	if p.digest == 0 {
+		digest := adler32.New()
```

Comment:
Instead of allocating an adler32 hash, can we use the zero-alloc variant we have?
https://github.com/m3db/stackadler32#stackadler32

Reply:
Here's an example of us using it (we actually use constructor from `src/dbnode/digest` I guess so we can swap later if we want to), you need to make the digest equal the return of the `Update` function:
https://github.com/m3db/m3/blob/master/src/dbnode/client/session.go#L3105:L3118
```go
		var (
			d        = digest.NewDigest()
			expected = uint32(*checksum)
		)
		if merged := block.Segments.Merged; merged != nil {
			d = d.Update(merged.Head).Update(merged.Tail)
		} else {
			for _, s := range block.Segments.Unmerged {
				d = d.Update(s.Head).Update(s.Tail)
			}
		}
		if actual := d.Sum32(); actual != expected {
			return fmt.Errorf("block checksum is bad: expected=%d, actual=%d", expected, actual)
		}
```